### PR TITLE
#52 Monitor Pedals

### LINKED ISSUE #52: Monitor Pedals task
This is a fairly high priority ticket, where we need to create an RTOS task to sample the values of the pedals, process it (maybe low pass filter to remove noise), and then add it into a queue somewhere. I think that there is some improvements on sampling we can make, especially how we represent this data upstream to tasks that depend on it (motor controller torque commands)

### PR DESCRIPTION
**Pull request by Peyton-McKee**
pedal go brrr

closes #52

### CODE REVIEW

**Comment by nwdepatie** [comment]
lmao peyton ticket

File: Core/Src/monitor.c [hunk 1]
```diff
@@ -68,4 +68,47 @@ void vTempMonitor(void *pv_params)
...
+
+	static onboard_pedals_t sensor_data;
+	fault_data_t fault_data = {
+		.id = ONBOARD_PEDAL_FAULT,
+		.severity = DEFCON4
+	};
+
+	/* Handle ADC Data for two input accelerator value and two input brake value*/
+	ADC_HandleTypeDef *hadc1 = (ADC_HandleTypeDef *)pv_params;
+	uint32_t adc_data[4];
```
[the remark this answers is not in the document]
Comment (nwdepatie) [reply Core/Src/monitor.c:95]: Also double check resolution on STM ADC's, I know the teensy's were like 12 bit max, so we might be able to get away with `uint16_t`

**Review by nwdepatie — CHANGES_REQUESTED** [review]
Overall this is essentially exactly what I was looking for which is great. In a future ticket we can handle faulting conditions just to get this merged in ASAP. Just make the small changes I suggested and then we good

File: Core/Src/main.c [hunk 2]
```diff
@@ -148,6 +148,7 @@ int main(void)
 
   /* USER CODE BEGIN RTOS_THREADS */
   temp_monitor_handle = osThreadNew(vTempMonitor, &hi2c1, &temp_monitor_attributes);
+  pedal_monitor_handle = osThreadNew(vPedalMonitor, &hi2c2, &pedal_monitor_attributes);
```
[the remark this answers is not in the document]
Comment (Peyton-McKee) [reply Core/Src/main.c:151]: oh right

**Comment by nwdepatie** [comment]
This builds now, but at what cost

**Review by nwdepatie — APPROVED** [review]
Going to approve so that we can enable some other tickets

**Pull request merged**